docs/DeveloperGuide.md

```diff
@@ -4,26 +4,231 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Setting Up & Getting Started
+
+1. Fork the repo at https://github.com/AY2324S1-CS2113T-W11-1/tp.
+2. Clone the fork into your computer.
+
+## Design & Implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+<img src="images/ArchitectureDiagram.png" />
+
+
+**Main components of SysLib Architecture**
+
+SysLib currently consists of four main components:
+
+- `UI`: User Interaction
+- `Parser`: Parsing User Response 
+- `Command`: Command Executor
+- `Data`: Holds the data of SysLib
```

Comment:
Would it be better to indicate where other packages (e.g., `storage`, `common`, etc.) belong?